tests/plugins/test_polars_lazyframe_extensions.py

```diff
@@ -0,0 +1,177 @@
+import pathlib
+import sys
+
+import polars as pl
+import pytest
+from polars.testing import assert_frame_equal
+
+from hamilton.plugins.polars_extensions import (
+    PolarsAvroReader,
+    PolarsAvroWriter,
+    PolarsCSVWriter,
+    PolarsDatabaseReader,
+    PolarsDatabaseWriter,
+    PolarsFeatherWriter,
+    PolarsJSONReader,
+    PolarsJSONWriter,
+    PolarsParquetWriter,
+    PolarsSpreadsheetReader,
+    PolarsSpreadsheetWriter,
+)
+from hamilton.plugins.polars_lazyframe_extensions import (
+    PolarsScanCSVReader,
+    PolarsScanFeatherReader,
+    PolarsScanParquetReader,
+)
+
+
+@pytest.fixture
+def df():
+    yield pl.LazyFrame({"a": [1, 2], "b": [3, 4]})
+
+
+def test_lazy_polars_lazyframe_csv(df: pl.LazyFrame, tmp_path: pathlib.Path) -> None:
+    file = tmp_path / "test.csv"
+
+    writer = PolarsCSVWriter(file=file)
+    kwargs1 = writer._get_saving_kwargs()
+    writer.save_data(df)
+
+    reader = PolarsScanCSVReader(file=file)
+    kwargs2 = reader._get_loading_kwargs()
+    df2, metadata = reader.load_data(pl.LazyFrame)
+
+    assert PolarsCSVWriter.applicable_types() == [pl.DataFrame, pl.LazyFrame]
+    assert PolarsScanCSVReader.applicable_types() == [pl.LazyFrame]
+    assert kwargs1["separator"] == ","
+    assert kwargs2["has_header"] is True
+    assert_frame_equal(df.collect(), df2.collect())
+
+
+def test_lazy_polars_parquet(df: pl.LazyFrame, tmp_path: pathlib.Path) -> None:
+    file = tmp_path / "test.parquet"
+
+    writer = PolarsParquetWriter(file=file)
+    kwargs1 = writer._get_saving_kwargs()
+    writer.save_data(df)
+
+    reader = PolarsScanParquetReader(file=file, n_rows=2)
+    kwargs2 = reader._get_loading_kwargs()
+    df2, metadata = reader.load_data(pl.DataFrame)
```

Comment:
I think this should be lazy frame right? (it's inconsequential but it's a little confusing to read)
```suggestion
    df2, metadata = reader.load_data(pl.LazyFrame)
```